balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -199,12 +221,13 @@ func (b *pickfirstBalancer) UpdateClientConnState(state balancer.ClientConnState
 	if len(state.ResolverState.Addresses) == 0 && len(state.ResolverState.Endpoints) == 0 {
 		// Cleanup state pertaining to the previous resolver state.
 		// Treat an empty address list like an error by calling b.ResolverError.
-		b.state = connectivity.TransientFailure
+		b.connectivityState = connectivity.TransientFailure
 		b.closeSubConnsLocked()
 		b.addressList.updateAddrs(nil)
 		b.resolverErrorLocked(errors.New("produced zero addresses"))
 		return balancer.ErrBadResolverState
 	}
+	b.healthCheckingEnabled = state.ResolverState.Attributes.Value(enableHealthListenerKeyType{}) == enableHealthListenerValue
```

Comment:
Is the value really important here? Would changing `EnableHealthListener` to simply add a `struct{}{}` (by value and not pointer) and here check for non-nil work?

Reply:
Also, we probably want to do this after checking for valid config. Because otherwise, we could run into a case where we get invalid config and therefore end up returning `balancer.ErrBadResolverState`, but also end up changing the internal state of the balancer here.

Reply:
Why not use a boolean instead?